HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -428,6 +428,97 @@ def self.apply_hvac(hpxml, weather, convert_shared_systems)
+        next unless heating_system.heating_efficiency_percent.nil?
+
+        heating_system.heating_efficiency_percent = 0.98
+        heating_system.heating_efficiency_percent_isdefaulted = true
+      elsif [HPXML::HVACTypeStove, HPXML::HVACTypePortableHeater, HPXML::HVACTypeFixedHeater].include? heating_system_type
+        next unless heating_system.heating_efficiency_percent.nil?
+
+        if heating_system_fuel == HPXML::FuelTypeElectricity
+          heating_system.heating_efficiency_percent = 0.98
+        elsif [HPXML::FuelTypePropane, HPXML::FuelTypeNaturalGas, HPXML::FuelTypeOil,
+               HPXML::FuelTypeOil1, HPXML::FuelTypeOil2, HPXML::FuelTypeOil4, HPXML::FuelTypeOil5or6,
+               HPXML::FuelTypeKerosene, HPXML::FuelTypeDiesel].include? heating_system_fuel
+          heating_system.heating_efficiency_percent = 0.84 # https://www.yankeedoodleinc.com/stoves/gas/ and https://www.kumastoves.com/Store/ProductDetails/arctic
+        elsif heating_system_fuel == HPXML::FuelTypeWoodCord
+          heating_system.heating_efficiency_percent = 0.60 # HEScore assumption
+        elsif heating_system_fuel == HPXML::FuelTypeWoodPellets
+          heating_system.heating_efficiency_percent = 0.78 # HEScore assumption
+        elsif [HPXML::FuelTypeCoal, HPXML::FuelTypeCoalAnthracite, HPXML::FuelTypeCoalBituminous, HPXML::FuelTypeCoke].include? heating_system_fuel
+          heating_system.heating_efficiency_percent = 0.75 # http://www.stanleycbierly.com/projects/the-warmth-and-economy-of-owning-a-coal-stove/
+        end
+        heating_system.heating_efficiency_percent_isdefaulted = true
+      elsif heating_system_type == HPXML::HVACTypeFireplace
+        next unless heating_system.heating_efficiency_percent.nil?
+
+        if heating_system_fuel == HPXML::FuelTypeElectricity
+          heating_system.heating_efficiency_percent = 0.98
+        elsif [HPXML::FuelTypeNaturalGas, HPXML::FuelTypePropane, HPXML::FuelTypeOil,
+               HPXML::FuelTypeOil1, HPXML::FuelTypeOil2, HPXML::FuelTypeOil4, HPXML::FuelTypeOil5or6,
+               HPXML::FuelTypeKerosene, HPXML::FuelTypeDiesel].include? heating_system_fuel
+          heating_system.heating_efficiency_percent = 0.80 # with inserts; https://www.volunteerenergy.com/energy-efficient-fireplace/
+        elsif [HPXML::FuelTypeWoodCord, HPXML::FuelTypeWoodPellets].include? heating_system_fuel
+          heating_system.heating_efficiency_percent = 0.70 # https://www.houselogic.com/organize-maintain/home-maintenance-tips/wood-burning-fireplace-inserts-save-energy/
+        elsif [HPXML::FuelTypeCoal, HPXML::FuelTypeCoalAnthracite, HPXML::FuelTypeCoalBituminous, HPXML::FuelTypeCoke].include? heating_system_fuel
+          heating_system.heating_efficiency_percent = 0.75 # assume coal-burning stove efficiency as oil-burning fireplace efficiency
+        end
+        heating_system.heating_efficiency_percent_isdefaulted = true
```

Comment:
I'm still not comfortable with this. Let's discuss.

Reply:
I would like to clarify that the shipment-weighted default lookup table provides efficiencies of natural gas wall furnaces. As floor furnaces are similar to wall furnaces, floor furnaces are mapped to wall furnaces. All fuel types are mapped to natural gas for wall/floor furnaces because the lookup table only provides efficiencies for natural gas wall/floor furnaces.

I did a little more research on the stoves and fireplaces. It was found that the difference in efficiency between stoves and fireplaces was not exceptionally large (~3%) for most manufacturers. I propose to use the same default efficiencies for stoves and fireplaces. 
The propane stoves/fireplaces tend to have slightly higher efficiency compared to natural gas stoves/fireplaces, but its degree is so small that it is negligible. As stoves/fireplaces that use fuel other than electricity, wood, wood pellets, propane, natural gas are quite uncommon, all other fuel types are mapped to natural gas or propane.